aip/general/0121.md

```diff
@@ -53,6 +53,19 @@ database schema. In fact, having an API that is identical to the underlying
 database schema is actually an anti-pattern, as it tightly couples the surface
 to the underlying system.
 
+### Resource requirements
+
+For a valid resource-oriented API, the following must be satisfied:
+
+- The resource must have a unique [resource name][] by which it can be
+  identified and referenced.
+- The resource exposes only standard methods ([Get][], [List][], [Create][],
+  [Update][], [Delete][]) or [custom methods][].
+- The resource API must support [Get][] and [List][] operations.
+- Requests must be stateless: any individual request is independent of
```

Comment:
I would not use the term `stateless`, that directly conflicts with language in https://google.aip.dev/136#stateless-methods.

Reply:
This was actually lifted from the "stateless protocol" section in this doc:

```
### Stateless protocol

As with most public APIs available today, resource-oriented APIs **must**
operate over a stateless protocol: The fundamental behavior of any individual
request is independent of other requests made by the caller.

In an API with a stateless protocol, the server has the responsibility for
persisting data, which may be shared between multiple clients, while clients
have sole responsibility and authority for maintaining the application state.
```

to be honest I feel like the guidance in this section is a little strange: a resource-oriented API necessitates some storage of the resource itself. I guess the point is that a single request doesn't depend on some other request, but IMO that's the default assumption vs something that needs to be called out specifically.

What do you think? Is it worth filing an issue to note some cleanup here?

Reply:
> but IMO that's the default assumption vs something that needs to be called out specifically.
>
> What do you think? Is it worth filing an issue to note some cleanup here?

Nah, I don't think it's worth a change atm. I'd rather we be explicit about the "stateless protocol" thing. If anything, I'd say perhaps we need clarification that this doesn't mean the same thing as AIP-136 Stateless Methods haha

> This was actually lifted from the "stateless protocol" section in this doc:

Ah indeed. Can you use "stateless protocol" consistently here then?

Reply:
SGTM! done.